eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/ExternalDependencyExposedCheck.java

```diff
@@ -0,0 +1,236 @@
+            if (ast.getType() == TokenTypes.PARAMETER_DEF) {
+                invalidParameterTypesMap.putAll(getInvalidTypeFromTypeArguments(ast.findFirstToken(TokenTypes.TYPE)));
+            }
+        }
+        return invalidParameterTypesMap;
+    }
+
+    /**
+     *  A helper function that checks TYPE AST node. Since both return type and input parameter argument type has
+     *  TYPE AST node under. This function applied to both.
+     *
+     * @param typeToken TYPE AST node
+     * @return a map that maps all the invalid TYPE_ARGUMENT node and the type name
+     */
+    private Map<DetailAST, String> getInvalidTypeFromTypeArguments(DetailAST typeToken) {
+        final Map<DetailAST, String> invalidTypesMap = new HashMap<>();
+
+        if (typeToken == null) {
+            return invalidTypesMap;
+        }
+
+        final DetailAST identToken = typeToken.findFirstToken(TokenTypes.IDENT);
+        // if there is no IDENT token, implies the token is a default java type.
+        if (identToken == null) {
+            return invalidTypesMap;
+        }
+
+        final String typeName = identToken.getText();
+        // if does not exist in the classPathMap, that implies the type is java default types, such as int.
+        if (classPathMap.containsKey(typeName)) {
+            if (!isValidLibrary(classPathMap.get(typeName))) {
+                invalidTypesMap.put(typeToken, typeName);
+            } else {
+                // Checks multiple type arguments
+                final DetailAST typeArgumentsToken = typeToken.findFirstToken(TokenTypes.TYPE_ARGUMENTS);
+                if (typeArgumentsToken != null) {
+                    for (DetailAST ast = typeArgumentsToken.getFirstChild(); ast != null; ast = ast.getNextSibling()) {
+                        if (ast.getType() == TokenTypes.TYPE_ARGUMENT) {
+                            String invalidTypeName = getInvalidTypeNameFromTypeArgument(ast);
+                            if (invalidTypeName != null) {
+                                invalidTypesMap.put(ast, invalidTypeName);
+                            }
+                        }
+                    }
+                } else {
+                    // Checks single type argument
+                    final DetailAST typeArgumentToken = typeToken.findFirstToken(TokenTypes.TYPE_ARGUMENT);
+                    String invalidTypeName = getInvalidTypeNameFromTypeArgument(typeArgumentToken);
+                    if (invalidTypeName != null) {
+                        invalidTypesMap.put(typeArgumentToken, invalidTypeName);
+                    }
+                }
+            }
+        }
+        return invalidTypesMap;
+    }
+
+    /**
+     * Get invalid type name.
+     *
+     * @param typeArgumentToken TYPE_ARGUMENT AST node
+     * @return an invalid type name if it is an invalid library. Otherwise, returns null.
+     */
+    private String getInvalidTypeNameFromTypeArgument(DetailAST typeArgumentToken) {
```

Comment:
This method's logic is not limited to just typeArgumentTokens, you can reuse this method call in other places in your code to reduce the duplication of your checkstyle logic.